services/121-service/src/programs/program/program.service.ts

```diff
@@ -1117,14 +1130,81 @@ export class ProgramService {
 
     const filteredColumnDetails = this.filterUnusedColumn(columnDetails);
     const response = {
-      fileName: `selected-for-validation-list.csv`,
+      fileName: this.getExportFileName(ExportType.selectedForValidation),
       data: this.jsonToCsv(filteredColumnDetails),
     };
 
     return response;
   }
 
-  public filterUnusedColumn(columnDetails): object[] {
+  private async getDuplicatePhoneNumbers(programId: number): Promise<FileDto> {
+    const allConnections = await this.connectionRepository.find({
+      select: ['id', 'customData', 'fsp'],
+      relations: ['fsp'],
+      where: {
+        programsApplied: Equal([programId]), // This is NOT a robust filter for this program-id's PA. :(
+        customData: Not(IsNull()),
+      },
+    });
+
+    const duplicates = allConnections.filter(connection => {
+      const others = without(allConnections, connection);
+      const currentPaNumbers = compact([
+        connection.customData[CustomDataAttributes.phoneNumber],
+        connection.customData[CustomDataAttributes.whatsappPhoneNumber],
+      ]);
+
+      const hasDuplicateProgramNr = this.hasDuplicateCustomDataValues(
```

Comment:
ProgramNr?

Reply:
At first I had `*PhoneNumber` everywhere but I renamed them all to be a bit shorter...

But this is called "program" as it is a program-specific attribute/question.
The PA-property "phoneNumber" is a more generic property, but currently these 2 'columns' are always the same, so I've settled on using this one only.

Reply:
Aha.. but then maybe ProgramPhoneNr? Because I was confused with ProgramId or something.